src/util.rs

```diff
@@ -20,6 +22,12 @@ pub(crate) trait BufExt2: Buf {
         arr
     }
 
+    fn get_vec_u8(&mut self, len: usize) -> Vec<u8> {
+        let mut ret = vec![0; len];
+        self.copy_to_slice(ret.as_mut_slice());
+        ret
+    }
+
```

Comment:
Does this have the same behaviour as all of the other functions in `BufExt2`? For example, does it consume `len` bytes from the `Buf`?